### PR TITLE
Milestone 2. Decorate class

### PR DESCRIPTION
**Pull request by indigodavid**
For this milestone, I implemented the following features:

- Add Nameable class and make Person class inherit from it
- Add Decorator base class that inherits from Nameable
- Add CapitalizeDecorator class that capitalizes the name of its Nameable component
- Add TrimmerDecorator class that trims the name of its Nameable component if it's bigger than 10 characters
- Test Decorators in school_library.rb

### CODE REVIEW

File: lib/school_library/capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'Decorator'
```
Comment (medaminedev66) [inline lib/school_library/capitalize_decorator.rb]: - [x] The file name is not capitalized. Please change `Decorator` to `decorator`.

File: lib/school_library/decorator.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'Nameable'
```
Comment (medaminedev66) [inline lib/school_library/decorator.rb]: - [x] The name of the file is not capitalized. Please change `Namebale` to `namebale`.

File: lib/school_library/person.rb [hunk 3]
```diff
@@ -1,8 +1,11 @@
-class Person
+require_relative 'Nameable'
```
Comment (medaminedev66) [inline lib/school_library/person.rb]: - [x] The name of the file is not capitalized. The word "Nameable" should be changed to "nameable."

File: lib/school_library/trimmer_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require_relative 'Decorator'
```
Comment (medaminedev66) [inline lib/school_library/trimmer_decorator.rb]: - [x] There is no capitalization in the filename. Please change `Decorator` to `decorator`.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @indigodavid   😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your code organization is good

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by indigodavid** [comment]
@medaminedev66 Thank you for noticing those capitalizations on those statements. Best regards and happy coding!

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @indigodavid , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**